src/main/scala/lasp/hapi/service/latis2/Latis2Util.scala

```diff
@@ -61,10 +62,35 @@ object Latis2Util {
             params <- f.getRange match {
               case s: Scalar if !s.hasName("time") =>
                 List(getParameterMetadata(s)).sequence
-              case Tuple(vs) =>
+              case Tuple(vs)   =>
                 // We are assuming no empty tuples and no nested tuples.
                 vs.toList.traverse(getParameterMetadata(_))
-              case _         => List().pure[F]
+              case g: Function =>
+                val size = g.getMetadata("length").get.toInt
+                val name = g.getDomain.getName
+                val units = g.getDomain.getMetadata("units").get
+                // Need to read the first sample in order to get the
+                // domain of the inner function.
+                val bin = it.next match {
+                  case Sample(_, Function(it)) =>
+                    val ws = it.map {
+                      case Sample(w, _) =>
+                        w.getData.asInstanceOf[NumberData].doubleValue
```

Comment:
I think you can just put `Number(d)` in the pattern match in place of "w" to get your double value.